tinygrad/tensor.py

```diff
@@ -87,6 +87,11 @@ def __init__(self, data, gpu=None, requires_grad=True):
   def __repr__(self):
     return f"Tensor {self.data!r} with grad {(self.grad.data if self.grad else None)!r}"
 
+  def __getitem__(self, index):
+    if not isinstance(index, (int, list, tuple)):
+      raise ValueError("{type(index)} not a valid index.")
+    return self.data[index] if not self.gpu else self.cpu().data[index]
```

Comment:
That can't convert it to cpu. If you want indexing, it has to work on the gpu too. It should also return Tensor type, not data.

Reply:
Sorry i though i pushed that 
Pushing it

Reply:
@geohot 
so for gpu
def gpu_index(index):
# Index GpuBuffer 
# return a new GPUBuffer with index data.
right?